ampd/src/handlers/stellar_verify_msg.rs

```diff
@@ -304,10 +304,12 @@ mod tests {
             metadata: PollMetadata {
                 poll_id: "100".parse().unwrap(),
                 source_chain: "stellar".parse().unwrap(),
-                source_gateway_address: ScAddress::Contract(stellar_xdr::curr::Hash::from([1; 32]))
-                    .to_string()
-                    .try_into()
-                    .unwrap(),
+                source_gateway_address: ScAddress::Contract(stellar_xdr::curr::ContractId(
+                    stellar_xdr::curr::Hash::from([1; 32]),
+                ))
```

Comment:
```suggestion
                source_gateway_address: ScAddress::Contract(stellar_xdr::curr::Hash::from([1; 32]).into())
```
adjust accordingly in other places
you could even do (maybe, I don't know if the compiler will be able to find this)
```suggestion
                source_gateway_address: ScAddress::Contract([1; 32].into().into())
```

Reply:
got it, updated to `source_gateway_address: ScAddress::Contract(stellar_xdr::curr::Hash::from([1; 32]).into())`. the latter give a compile time error.